metadatastore/commands.py

```diff
@@ -604,48 +953,18 @@ def find_run_starts(**kwargs):
     ...                stop_time=time.time())
 
     """
-    _normalize_object_id(kwargs, '_id')
+    run_start = kwargs.pop('run_start', None)
+    if run_start:
+        run_start = doc_or_uid_to_uid(run_start)
+        run_start = runstart_given_uid(run_start)
+        run_start = _RUNSTART_UID_to_OID_MAP[run_start['uid']]
+        kwargs['_id'] = run_start
+
     _format_time(kwargs)
 
-    rs_objects = RunStart.objects(__raw__=kwargs).order_by('-time')
-    rs_objects = rs_objects.no_dereference()
-    _as_document = _AsDocument()
-    return (_as_document(rs) for rs in rs_objects)
+    rs_objects = RunStart.objects(__raw__=kwargs).as_pymongo()
 
-
-@_ensure_connection
-def find_beamline_configs(**kwargs):
-    """Given search criteria, locate BeamlineConfig Documents.
-
-    Parameters
-    ----------
-    start_time : time-like, optional
-        time-like representation of the earliest time that a BeamlineConfig
-        was created. Valid options are:
-           - timestamps --> time.time()
-           - '2015'
-           - '2015-01'
-           - '2015-01-30'
-           - '2015-03-30 03:00:00'
-           - datetime.datetime.now()
-    stop_time : time-like, optional
-        timestamp of the latest time that a BeamlineConfig was created. See
-            docs for `start_time` for examples.
-    uid : str, optional
-        Globally unique id string provided to metadatastore
-    _id : str or ObjectId, optional
-        The unique id generated by mongo
-
-    Returns
-    -------
-    beamline_configs : iterable of metadatastore.document.Document objects
-    """
-    _format_time(kwargs)
-    # ordered by _id because it is not guaranteed there will be time in cbonfig
-    beamline_configs = BeamlineConfig.objects(__raw__=kwargs).order_by('-_id')
-    beamline_configs = beamline_configs.no_dereference()
-    _as_document = _AsDocument()
-    return (_as_document(bc) for bc in beamline_configs)
+    return (_cache_runstart(rs) for rs in rs_objects.order_by('-time'))
 
 
 @_ensure_connection
```

Comment:
The docstring for the run_start parameter needs to be changed to metadatastore.doc.Document